kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/EventHandlerServlet.java

```diff
@@ -81,14 +88,20 @@ public void handleEvent(final Event event) {
     public void doGet(final HttpServletRequest request, final HttpServletResponse response)
             throws ServletException, IOException {
 
+        // if the queue is full, clear it as we don't need to care about rejected events
+        if (this.sharedCache.size() == MAX_SIZE_OF_QUEUE) {
+            this.sharedCache.clear();
```

Comment:
Let's suppose there does not exist any consumer thread but producers are producing event data continuously. So, max of 10 elements will be there in the queue. Now, a consumer thread appears and it now calls `doGet()` on the servlet. According to the requirement of this feature, the consumer thread is not responsible to consume stale event data that are already in the queue. It, hence, first clears the queue and waits for new event data so that it can consume thereafter.

The requirement is that as soon as an user creates a Kura Wires `Wire Graph` and deploys it in the runtime, the data produced by Producer nodes in the `Wire Graph` will fire an event which will be reflected in the `WireGraph` in real-time.  The connections or arrows in the `Wire Graph` will glow as the consumer thread simultaneously is consuming data from the queue.

In summary,

Producer node or `Wire Component` produces data -> it fires an event -> the event is inserted in the queue -> the consumer consumes the data from the queue -> the arrow in the `Wire Graph` glows